mat/band.go

```diff
@@ -126,7 +127,10 @@ func (t TransposeBand) UntransposeBand() Banded {
 // which is passed to NewBandDense as []float64{*, 1, 2, 3, 4, ...} with kl=1 and ku=2.
 // Only the values in the band portion of the matrix are used.
 func NewBandDense(r, c, kl, ku int, data []float64) *BandDense {
-	if r < 0 || c < 0 || kl < 0 || ku < 0 {
+	if r <= 0 || c <= 0 || kl < 0 || ku < 0 {
```

Comment:
I think I'd go with less indentation:
```go
switch {
case r == 0 || c == 0:
    panic(ErrZeroLength)
case r < 0 || c < 0 || kl < 0 || ku < 0:
    panic("mat: negative dimension")
}
```

(same for the other nested `if`s)
but ok.

Reply:
The choice to use repeated checks was made to reduce compares on the fast path. The use of `switch` would undo that.